src/main/webapp/script.js

```diff
@@ -29,34 +29,33 @@ function loadTasks() {
 }
 
 function renderTasks(tasks) {
-  $("#task-list").empty();
+  $("#task-list > .task").remove();
   $("#schedule-button").prop("disabled", !tasks.length);
   $("#empty-message").toggle(!tasks.length);
   if (tasks.length > 0) {
     tasks.forEach(renderSingleTask);
   }
+  initDropdowns();
 }
 
+/**
+ * Compiles the task template with the actual task data and
+ * adds it to the task list.
+ */
 function renderSingleTask(task) {
-  id = task.id;
-  $("#task-list").append(
-    `<li>
-      <label class="mdc-list-item" role="checkbox" aria-checked="false">
-        <span class="mdc-list-item__ripple"></span>
-        <span class="mdc-list-item__graphic">
-          <div class="mdc-checkbox">
-            <input type="checkbox" name="taskId" id="${id}" class="mdc-checkbox__native-control" value="${id}" />
-            <div class="mdc-checkbox__background">
-              <svg class="mdc-checkbox__checkmark" viewBox="0 0 24 24">
-                <path class="mdc-checkbox__checkmark-path" fill="none" d="M1.73,12.91 8.1,19.28 22.79,4.59"/>
-              </svg>
-              <div class="mdc-checkbox__mixedmark"></div>
-            </div>
-          </div>
-        </span>
-        <label class="mdc-list-item__text" for="${id}">${task.title}</label>
-      </label>
-    </li>`);
+  const taskTemplateElement = document.getElementById("task-template").innerHTML;
```

Comment:
These 2 steps (retrieving and compiling the template) should be done once, at the startup, and not when adding every task.
If done at the startup, you could probably also keep the `$("#task-list").empty();` since the template function would be compiled and preserved for later use.